packages/tools/devtools/devtools-core/src/data-visualization/DataVisualization.ts

```diff
@@ -482,8 +518,30 @@ export async function visualizeChildData(
  * Determines whether or not the provided value is an {@link ISharedObject}, for the purposes of this library.
  * @remarks Implemented by checking for the particular properties / methods we use in this module.
  */
-function isSharedObject(value: Partial<ISharedObject>): value is ISharedObject {
+function isSharedObject(value: unknown): value is ISharedObject {
 	return (
-		value.id !== undefined && value.attributes?.type !== undefined && value.on !== undefined
+		(value as SharedObject).id !== undefined &&
+		(value as SharedObject).attributes?.type !== undefined &&
+		(value as SharedObject).on !== undefined
 	);
 }
+
+/**
+ * Determines whether or not the provided value is an {@link DataObject} using `instanceof`, for the purposes of this library.
+ * @remarks `instanceof` is preferred over checking specific properties or methods, because a version mix-up with
+ * @remarks {@link @fluidframework/aqueduct#} is unlikely in devtools and end-user applications.
+ * @remarks `value` is type casted as a workaround to access the `root` property of the {@link DataObject} (private).
+ */
+function isDataObject(value: unknown): value is DataObject {
+	if (value instanceof DataObject === false) {
+		return false;
+	}
```

Comment:
Just calling out that if you follow the local-linking approach that we talked about in standup today, and only link the new version of Devtools but not of every other Fluid package, a DataObject probably won't make it through this check and won't visualize correctly :).

Reply:
Good point. I can revert back to the check attribute / properties for now, and once the testing's done we can revert back into `instanceof` method.

Reply:
We could leave the check here but only for the positive case; if `instanceof` returns true, we know for sure that the thing is a DataObject. If not, it could be that it's really not, or a version mismatch, and in that case we could check for the properties, and if they seem to be what we expect, still return true.